client/package.json

```diff
@@ -5,12 +5,16 @@
   "description": "cellxgene is a web application for the interactive exploration of single cell sequence data.",
   "repository": "https://github.com/chanzuckerberg/cellxgene",
   "scripts": {
-    "clean": "rimraf build",
     "build": "npm run clean && webpack --config",
-    "prod": "npm run build -- configuration/webpack/webpack.config.prod.js",
+    "clean": "rimraf build",
     "dev": "npm run build -- configuration/webpack/webpack.config.dev.js",
-    "fmt": "eslint --fix src",
-    "lint": "eslint src"
+    "e2e": "jest --config __tests__/e2e/e2eJestConfig.json e2e/e2e.test.js",
```

Comment:
resorted object keys alphabetically and added extra commands `e2e`, `e2e-annotations`, and `e2e-prod` for ease of development

Reply:
This duplicates the code that runs the test suites. The other copies are in `client/Makefile`. It makes sense to me to pick one to be authoritative: either make or npm scripts. This is more DRY and reduces the chance that some future developer will run the test one way, it passes, and then it goes out to CI, is run with a different entrypoint, and fails.

Some history: I refactored these a while ago and (showing my BE-engineering bias a bit here here) I put them all in make. This had the advantage of being able to parameterize runs of node with environment variables from `environment.default`, locking everything together.

Reply:
Thanks for the great idea, @mweiden ! I just updated `client/Makefile` to use the npm scripts:

E.g.,

```
.PHONY: e2e
e2e:
	JEST_ENV=${PROD} \
	CXG_CLIENT_PORT=${CXG_SERVER_PORT} \
		npm run e2e -- \
		--verbose false \

.PHONY: e2e-annotations
e2e-annotations:
	JEST_ENV=${PROD} \
	CXG_CLIENT_PORT=${CXG_SERVER_PORT} \
	npm run e2e-annotations -- \
		--verbose false \
```
 -- 

I only duplicated some vars from `environment.default` to `e2e/config.js` because I wanted to set:

```ts
export const appPort = process.env.CXG_CLIENT_PORT || CXG_CLIENT_PORT;

export const appUrlBase =
  process.env.CXG_URL_BASE || `http://localhost:${appPort}`;
```

This allows FE engineers to optionally run npm scripts locally with add-on tools like [`ndb`](https://github.com/GoogleChromeLabs/ndb) for debugging, and extra flags, like `-u` to update snapshots and `HEADFUL=true` to run tests in headful mode regardless of `JEST_ENV` value (prod, dev, debug)

But I was thinking maybe turning `environment.default` into `environment.default.json` would mean both Python and JS could share the same source of truth 😄 ?

Thank you!

Reply:
Using `environment.default.json` now 🎉 !

Reply:
Looks better!